pkg/storage/local_test.go

```diff
@@ -213,6 +214,711 @@ func TestGetReferrers(t *testing.T) {
 	})
 }
 
+func FuzzNewBlobUpload(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data string) {
+		dir := t.TempDir()
+		defer os.RemoveAll(dir)
+		t.Logf("Input argument is %s", data)
+		log := log.Logger{Logger: zerolog.New(os.Stdout)}
+		metrics := monitoring.NewMetricsServer(false, log)
+		imgStore := storage.NewImageStore(dir, true, storage.DefaultGCDelay, true, true, log, metrics)
+
+		_, err := imgStore.NewBlobUpload(data)
+		if err != nil {
+			if isKnownErr(err) {
+				return
+			}
+
+			t.Error(err)
+		}
+	})
+}
+
+func FuzzPutBlobChunk(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data string) {
+		dir := t.TempDir()
+		defer os.RemoveAll(dir)
+		t.Logf("Input argument is %s", data)
```

Comment:
Can you briefly explain how the go toolchain identifies the correct data in the separate repo for a specific test?
What is the repo used for? I was under the impression the data was supposed to be dynamically generated for each test execution.

Reply:
The Go toolchain does not identify data from a separate repo. Go just expects a folder named 'testdata' to be present inside the package it tests, before fuzzing starts.
That's why we have these 2 lines in the newly added fuzz-all Make target:
```
git clone https://github.com/project-zot/test-data.git; \
mv test-data/storage pkg/storage/testdata; \
```
As fuzz tests are created for different functionalities/packages, the fuzz-all Make target needs completion with new "mv test-data/blabla pkg/blabla/testdata" commands.

The separate repo exists for fuzz data to be gathered, since it can get bigger over time.

See https://github.com/project-zot/zot/blob/f72e4dfdff2187e444781f217816cf79e19c0758/README_fuzz.md for a better view.

Reply:
Got it.
What about having the fuzz tests in the same file with the other tests? What happens when we run the usual go test command? Are they skipped because the data is missing? Do they count for coverage in any way?

Reply:
Native Go fuzzing encourages having Fuzz tests in the same file with the other tests. The usual go test command won't run fuzz test without the "-fuzz" flag, followed by the name of the test to fuzz.
As I've seen , they won't count for coverage. Maybe using the -cover flag (as used with usual go test) will make Fuzz tests count.

Reply:
Looking at https://go.dev/doc/fuzz/

```
There are two modes of running your fuzz test: as a unit test (default go test), or with fuzzing (go test -fuzz=FuzzTestName).

Fuzz tests are run much like a unit test by default. Each [seed corpus entry](https://go.dev/doc/fuzz/#glos-seed-corpus) will be tested against the fuzz target, reporting any failures before exiting.
```
Since the seed corpus would not normally be available (since the test-data folders are not normally copied under the repo), are these not run at all?

Reply:
I think we should be fine.

Reply:
Yes, you are right. Without "-fuzz", they will be run as unit tests. Without input data, they will finish instantly, as if skiped.